frontend/www/js/omegaup/groups.ts

```diff
@@ -1,18 +1,91 @@
+import { types } from './api_types';
 import T from './lang';
+import * as ui from './ui';
+import * as CSV from '@/third_party/js/csv.js/csv.js';
 
-export function cleanRecords(
-  records: (null | number | string)[][],
-): (undefined | string)[][] {
-  return records.map((row) =>
-    row.map((cell) => {
-      if (cell === null) {
-        return undefined;
-      }
-      if (typeof cell !== 'string') {
-        return String(cell);
+export function downloadCsvFile({
+  fileName,
+  columns,
+  records,
+}: {
+  fileName: string;
+  columns: string[];
+  records: types.Identity[];
+}): void {
+  const dialect = {
+    dialect: {
+      csvddfVersion: 1.2,
+      delimiter: ',',
+      doubleQuote: true,
+      lineTerminator: '\r\n',
+      quoteChar: '"',
+      skipInitialSpace: true,
+      header: true,
+      commentChar: '#',
+    },
+  };
+  const fields = columns.map((column) => ({ id: column }));
+  const csv = CSV.serialize({ fields, records }, dialect);
+  const hiddenElement = document.createElement('a');
+  hiddenElement.href = `data:text/csv;charset=utf-8,${window.encodeURIComponent(
+    csv,
+  )}`;
+  hiddenElement.target = '_blank';
+  hiddenElement.download = fileName;
+  hiddenElement.click();
+}
+
+export function getCSVRecords<
+  T extends Record<string, string | number | undefined>
```

Comment:
```suggestion
  T extends object
```

perdón, esto fue mi culpa: había probado con `type Foo = { ... };` en vez de `interface Foo { ... };`, que resulta que son _ligeramente_ distintos para TypeScript. ahora sabemos que lo son!

Reply:
Esto es muy revoltoso. Voy a seguir repasando el handbook de typescript para entender mejor esta parte. 

y parece que al compilador no le gustó usar `object`, me marca este error:
```
frontend/www/js/omegaup/groups.ts:38:41: [eslint] Don't use `object` as a type. The `object` type is 
currently hard to use ([see this issue](https://github.com/microsoft/TypeScript/issues/21732)).
Consider using `Record<string, unknown>` instead, as it allows you to more easily inspect and use the keys. 
[@typescript-eslint/ban-types]
    export function getCSVRecords<T extends object>({
```

Y si utilizo la sugerencia que proponen, vuelvo a obtener el error de antes:
```
Type 'Identity' does not satisfy the constraint 'Record<string, unknown>'.
```
Lo único que he encontrado para arreglarlo es usar
```js
T extends Record<string, any>
```
, pero no sé si sea correcto utilizarlo en este contexto.

Lo subiré como propones tú para que me ayudes a revisar como deberá quedar porfa.